stacks-core/src/crypto/hash.rs

```diff
@@ -0,0 +1,77 @@
+use ring::digest::Context;
+use ring::digest::SHA256;
+use ripemd::Digest;
+use ripemd::Ripemd160;
+
+const SHA256_LENGTH: usize = 32;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SHA256Hash([u8; SHA256_LENGTH]);
+
+impl SHA256Hash {
+    pub fn new(value: impl AsRef<[u8]>) -> Self {
+        let bytes = {
+            let mut buff = [0u8; SHA256_LENGTH];
+
+            let mut ctx = Context::new(&SHA256);
+            ctx.update(value.as_ref());
+
+            let digest = ctx.finish();
+            buff.copy_from_slice(digest.as_ref());
```

Comment:
Would it be possible to avoid copying here?

Reply:
I've now replaced all occurences with `<slice>.try_into()`. It's suggested as the best way to do this and the compiler will probably be able to optimize it as much as possible.

Reply:
The only way to get rid of copying is to change types to borrow data, which might be interesting but potentially complicated. I don't think we need to care that much about performance.